pkg/geo/geomfn/unary_operators.go

```diff
@@ -112,3 +112,39 @@ func perimeterFromGeomT(geomRepr geom.T) (float64, error) {
 func Area(g *geo.Geometry) (float64, error) {
 	return geos.Area(g.EWKB())
 }
+
+// Dimension returns the topological dimension of a given Geometry.
+func Dimension(g *geo.Geometry) (int, error) {
+	t, err := g.AsGeomT()
+	if err != nil {
+		return 0, err
+	}
+	return dimensionFromGeomT(t)
+}
+
+// dimensionFromGeomT returns the dimension from a geom.T, recursing down
+// GeometryCollections if required.
+func dimensionFromGeomT(geomRepr geom.T) (int, error) {
+	switch geomRepr := geomRepr.(type) {
```

Comment:
can we use the denormalised `geo.Geometry.ShapeType()` function to do this, to avoid the conversion?

Reply:
ah GeometryCollection would be a pain, yeah let's keep this.